iterable-extension/src/test/java/com/mparticle/ext/iterable/IterableExtensionTest.java

```diff
@@ -548,6 +607,92 @@ public void testProcessAudienceMembershipChangeRequest() throws Exception {
         assertEquals(3, i);
     }
 
+    @Test
+    public void testProcessAudienceMembershipChangeWithMPID() throws IOException {
+        testIterableExtension.iterableService = iterableServiceMock;
+        Mockito.when(iterableServiceMock.listSubscribe(Mockito.any(), Mockito.any()))
+                .thenReturn(callMock);
+        Mockito.when(iterableServiceMock.listUnsubscribe(Mockito.any(), Mockito.any()))
+                .thenReturn(callMock);
+        Mockito.when(callMock.execute()).thenReturn(testSuccessResponse);
+
+        List userProfileList = new LinkedList<UserProfile>();
+        testUserProfile.setMpId("m1");
+        testUserProfile.setAudiences(testAudienceList);
+        testUserProfileWithEmail.setMpId("m2");
+        testUserProfileWithEmail.setAudiences(testAudienceList);
+        testUserProfileWithEmailAndCustomerId.setMpId("m3");
+        testUserProfileWithEmailAndCustomerId.setAudiences(testAudienceList);
+        userProfileList.add(testUserProfile);
+        userProfileList.add(testUserProfileWithEmail);
+        userProfileList.add(testUserProfileWithEmailAndCustomerId);
+
+        AudienceMembershipChangeRequest request = new AudienceMembershipChangeRequest();
+        testAccount.getAccountSettings().put(SETTING_USER_ID_FIELD, USER_ID_FIELD_MPID);
+        request.setAccount(testAccount);
+        request.setUserProfiles(userProfileList);
+
+        testIterableExtension.processAudienceMembershipChangeRequest(request);
+
+        ArgumentCaptor<SubscribeRequest> subscribeArgs = ArgumentCaptor.forClass(SubscribeRequest.class);
+        List<SubscribeRequest> subscribeRequests = subscribeArgs.getAllValues();
+        Mockito.verify(iterableServiceMock, Mockito.times(2)).listSubscribe(Mockito.any(), subscribeArgs.capture());
+        Collections.sort(
+                subscribeRequests,
+                (a, b) -> a.listId > b.listId ? 1 : a.listId == b.listId ? 0 : -1
+        );
+        assertEquals(new Integer(1), subscribeRequests.get(0).listId);
```

Comment:
Hmm, is `new Integer(...)` necessary? Does it fail with just 1?

Reply:
The compiler was complaining since [SubscribeRequest](https://github.com/Iterable/mparticle-firehose-iterable/blob/ef59a8adbf6de52a4075887447b2dda51a9fbc94/iterable-java-sdk/src/main/java/com/mparticle/iterable/SubscribeRequest.java#L6) defines listId as class `Integer`. Does this read better?
```
assertEquals(1, subscribeRequests.get(0).listId.intValue());
```

Reply:
I updated it to use my suggestion. Probably a nit but seems better to compare primitive types.